src/content_sao.cpp

```diff
@@ -671,6 +661,16 @@ int LuaEntitySAO::punch(v3f dir,
 		m_env->getScriptIface()->luaentity_on_death(m_id, puncher);
 	}
 
+	char log[256];
+	std::string desc_puncher = puncher->getDescription(),
+			desc_self = getDescription();
+	porting::mt_snprintf(log, sizeof(log),
```

Comment:
why not just stream instead of using snprintf here ?

Reply:
IMO, snprintf results in a formatting which is much easier to understand. Or would you prefer a mess like this?
`actionstream << desc_puncher << " (id=" << puncher->getId() << ", hp=" << puncher->getHP() << ") punched " << desc_self << " (id=" << m_id << ", hp=" << m_hp << "), damage=" << (old_hp - (s32)getHP()) << damage_handled ? " (handled by Lua)" : "") << std::endl;`

Reply:
it's the correct way, you want truncate the log here, description can be very high in terms of characters, same for puncher description. If it's a problem we should consider using a logging library like log4cpp or log4cplus which is more configurable and permits a such pattern without problems